src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,116 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import { TodoStatus } from '../../types';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+import { TodoFilter } from '../TodoFilter';
+
+interface Props {
+  todos: Todo[];
+  selectUser: (userId: number) => void;
+}
+
+interface State {
+  titlePhrase: string;
+  todoStatus: TodoStatus;
+}
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    titlePhrase: '',
+    todoStatus: TodoStatus.Default,
+  };
+
+  handleChange = (value: string) => {
+    this.setState({
+      titlePhrase: value,
+    });
+  };
+
+  selectStatus = (status: TodoStatus) => {
+    this.setState({
+      todoStatus: status,
+    });
+  };
+
+  getTodoStatus = (completed: boolean) => {
+    const { todoStatus } = this.state;
+
+    switch (todoStatus) {
+      case TodoStatus.Active:
+        return !completed;
+      case TodoStatus.Completed:
+        return completed;
+      default:
+        return true;
+    }
+  };
+
+  getFilteredTodos = (todos: Todo[]) => {
+    const { titlePhrase } = this.state;
+
+    return todos.filter(
+      todo => {
+        const { title, completed } = todo;
+
+        return title && title.includes(titlePhrase) && this.getTodoStatus(completed);
```

Comment:
it will not find if the search is entered in uppercase. In such cases, it is better to add the ability to print in any case and find the results